hs_collection_resource/views.py

```diff
@@ -40,65 +40,103 @@ def update_collection(request, shortkey, *args, **kwargs):
-                                    needed_permission=ACTION_TO_AUTHORIZE.VIEW_METADATA)
-
-                    is_shareable = res_to_add.raccess.shareable
-                    is_owner = res_to_add.raccess.owners.filter(pk=user.pk).exists()
-                    if not is_shareable and not is_owner:
-                            raise Exception('Only resource owner can add a non-shareable '
-                                            'resource to a collection ')
-
-                    # add this new res to collection
-                    res_obj_add = get_resource_by_shortkey(res_id_add)
-                    collection_res_obj.resources.add(res_obj_add)
-
-                    # check loop here
-                    # not implemented
-
-                    # change "Relation" metadata in collection
-                    value = RES_LANDING_PAGE_URL_TEMPLATE.format(res_id_add)
-                    add_or_remove_relation_metadata(add=True, target_res_obj=collection_res_obj,
-                                                    relation_type=hasPart, relation_value=value,
-                                                    set_res_modified=False)
+            if update_type == "add":
+                res_id_list_add = updated_contained_res_id_list
+                for res_id_add in res_id_list_add:
+                    if res_id_add in res_id_list_current_collection:
+                        raise Exception('Cannot add resource {0} as it '
+                                        'is already contained in collection'.format(res_id_add))
+            elif update_type == "set":
+                for res_id_add in updated_contained_res_id_list:
+                    if res_id_add not in res_id_list_current_collection:
+                        res_id_list_add.append(res_id_add)
+            for res_id_add in res_id_list_add:
+                # check authorization for all new resources being added to the collection
+                # the requesting user should at least have metadata view permission for each of
+                # the new resources to be added to the collection
+                res_to_add, _, _ \
+                    = authorize(request, res_id_add,
+                                needed_permission=ACTION_TO_AUTHORIZE.VIEW_METADATA)
+
+                # the resources being added should be 'Shareable'
+                # or is owned by current user
+                is_shareable = res_to_add.raccess.shareable
+                is_owner = res_to_add.raccess.owners.filter(pk=user.pk).exists()
+                if not is_shareable and not is_owner:
+                        raise Exception('Only resource owner can add a non-shareable '
+                                        'resource to a collection ')
+
+                # add this new res to collection
+                res_obj_add = get_resource_by_shortkey(res_id_add)
+                collection_res_obj.resources.add(res_obj_add)
+
+                # check loop here
```

Comment:
@pkdash 
current "test_collection.py" has several test cases for this view function. will add more to include new "update_type": "add" and "remove"